Sprint 1

gloginov opened the pull request.

gloginov committed:
[practicum_1]
+init repo
+add docker
+install Handlebars
+add entrypoint
+create Navigation page
+create Layout with Handlebars
+create custom component

gloginov committed:
[practicum_1]
+add pages
+add some components

gloginov committed:
[practicum_1]
*add links on navigation page

gloginov committed:
[practicum_1]
*add links on navigation page

gloginov committed:
[practicum_1]
*change image url in mock json
*add name for input type file
*change import file paths
+add declaration for build
*add in Readme part for build in docker

gloginov committed:
[practicum_1]
*add netlify badge

gloginov committed:
[practicum_1]
*add netlify file redirect
*change gitignore

gloginov committed:
[practicum_1]
*add netlify file redirect

gloginov committed:
[practicum_1]
*add fov

gloginov committed:
[practicum_1]
*add pages in README.md

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

К сожалению, работу придется отклонить без потери итерации, так как автотесты должны проходить успешно. Скрин https://disk.yandex.ru/i/0UUsLWxpsTnuHw

gloginov committed:
[practicum_1]
*fix project for autotest

gloginov committed:
[practicum_1]
*add 'npm start'

gloginov committed:
[practicum_1]
*add new line in mixins.scss

gloginov committed:
[practicum_1]
*remove some style files

gloginov committed:
[practicum_1]
*new line

gloginov committed:
[practicum_1]
*new line

gloginov committed:
[practicum_1]
*fix miss quote

gloginov committed:
[practicum_1]
*fix miss quote

gloginov committed:
[practicum_1]
*add start script

mikebars1995 commented on components/selected-chat/selected-chat.hbs:
```diff
@@ -0,0 +1,61 @@
+                    {{#if (isText this.type)}}
+                        {{{ this.message }}}
+                    {{else if (isImage this.type)}}
+                        <img src={{ this.message }} alt="">
```
**Можно лучше**

Каждой картинке (тегу `img`) нужен информативный атрибут `alt` (например: имя картинки вставлять), так как при ошибке загрузки картинки на сайте подсветится **название** этой картинки (которое в `alt`). Еще это нужно для **плохо видящих** и `незрячих` людей - для них есть специальные средства, которые распознают текст в `alt` и проговаривают его. Подробнее тут https://habr.com/ru/post/341810/

gloginov replied:
Пока делать не стал, как будут данные (живые) станет понятно что можно подставить.

mikebars1995 commented on layouts/centered/centered.hbs:
```diff
@@ -0,0 +1,6 @@
+<div class="layout-center layout-center_{{ position }}">
```
**Нужно исправить**

По семантике нужно помещать весь главный контент страницы в тег `main`.


**Можно лучше**

Чтобы сделать сайт более доступным для пользователя, нужно стараться придерживаться **семантической верстки**.
Несколько полезных статей на эту тему:
•	[Что такое семантическая вёрстка и зачем она нужна](https://htmlacademy.ru/blog/html/semantics)
•	[Семантика для циников](https://www.youtube.com/watch?v=B5pomLr059U)

gloginov replied:
Т.к. этот layout используется в представлении модального окна, изменил тег div на main в index.html.

mikebars1995 commented on pages/profile/settings/setting.hbs:
```diff
@@ -0,0 +1,41 @@
+                </div>
+
+                <div class="profile-info">
+                    {{> TextFieldLabel label="Почта" type="text" errorMessage="" value="ivanivanov@mm.rr" name="login" className="profile-info-field inline "}}
```
**Нужно исправить**

По заданию 1го спринта нужно давать инпутам **конкретные** атрибуты `name`, чтобы в 3м спринте не менять ничего при подключении к `Api`. Скрин из задания https://disk.yandex.ru/i/lSPqeGrAa20ayQ

gloginov replied:
Сделал.

mikebars1995 commented on pages/profile/settings/setting.hbs:
```diff
@@ -0,0 +1,41 @@
+                    </label>
+                </div>
+
+                <div class="profile-info">
```
**Нужно исправить**


Инпуты и кнопку сабмита, по семантике, нужно обернуть в тег `form`, так как это форма

gloginov replied:
Сделал.

mikebars1995 commented on index.html:
```diff
@@ -0,0 +1,24 @@
+<!doctype html>
+<html lang="en">
```
**Можно лучше**


Лучше установить корректный язык страницы `<html lang="ru">`

Раз интерфейс проекта русскоязычный, то и декларация языка должна ему соответствовать. Некоторые устройства могут постоянно предлагать перевести сайт на другой язык, или ничего не спрашивая, тихо менять страницу. Вот одна из таких историй: https://habr.com/ru/post/650907/

gloginov replied:
Сделал.

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на *гитхабе* все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная


Что сделано отлично:


* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов
* Отлично, что цвета вынесены в `css`-константы
* Отлично, что моковые данные вынесены в отдельные файлы

Что нужно исправить:

* По семантике нужно помещать весь главный контент страницы в тег `main`.
* По заданию 1го спринта нужно давать инпутам **конкретные** атрибуты `name`, чтобы в 3м спринте не менять ничего при подключении к `Api`. Скрин из задания https://disk.yandex.ru/i/lSPqeGrAa20ayQ 
* Инпуты и кнопку сабмита, по семантике, нужно обернуть в тег `form`, так как это форма 

Что можно улучшить:

* Каждой картинке (тегу `img`) нужен информативный атрибут `alt` (например: имя картинки вставлять), так как при ошибке загрузки картинки на сайте подсветится **название** этой картинки (которое в `alt`). Еще это нужно для **плохо видящих** и `незрячих` людей - для них есть специальные средства, которые распознают текст в `alt` и проговаривают его. Подробнее тут https://habr.com/ru/post/341810/ 
* Лучше установить корректный язык страницы `<html lang="ru">`


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

**Комментарии `Можно лучше` не обязательны к исправлению прямо сейчас, это рекомендации**

Удачного рефакторинга кода.

gloginov committed:
[practicum_1]
*replace tag div to main for general mount container
*replace div to form on setting page
*change names for fields
*add type for Button

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
